WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift

```diff
@@ -875,6 +875,8 @@ private extension NotificationsViewController {
             return
         }
 
+        WPAnalytics.track(.notificationsMarkAllReadTapped)
```

Comment:
This can be moved on top of the `guard let` depending on the requirement. The button is disabled when there are no notifications or they are already all marked as read. But it could still happen that `fetchedObjects` return nil. Looking around some similar examples, I think not firing the event in the convention in such cases.

Reply:
Agreed. Thanks for explaining that!